app/frontend/src/AppRoute.tsx

```diff
@@ -0,0 +1,90 @@
+import { Container, makeStyles } from "@material-ui/core";
+import React, { useEffect } from "react";
+import { Redirect, Route, RouteProps } from "react-router-dom";
+
+import Navigation from "./components/Navigation";
+import { useAuthContext } from "./features/auth/AuthProvider";
+import { jailRoute, loginRoute } from "./routes";
+
+const useStyles = makeStyles((theme) => ({
+  standardContainer: {
+    paddingBottom: theme.shape.navPaddingMobile,
+    paddingLeft: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    [theme.breakpoints.up("md")]: {
+      paddingBottom: 0,
+      paddingTop: theme.shape.navPaddingDesktop,
+    },
+  },
+  fullscreenContainer: {
+    padding: 0,
+    margin: 0,
+  },
+}));
+
+interface AppRouteProps extends RouteProps {
+  isPrivate?: boolean;
+  isFullscreen?: boolean;
+}
+
+export default function AppRoute({
+  children,
+  isPrivate = true,
+  isFullscreen = false,
+  ...otherProps
+}: AppRouteProps) {
+  const { authState, authActions } = useAuthContext();
+  const isAuthenticated = authState.authenticated;
+  const isJailed = authState.jailed;
+  useEffect(() => {
+    if (!isAuthenticated && isPrivate) {
+      authActions.authError("Please log in.");
+    }
+  });
+
+  const classes = useStyles();
+
+  if (isPrivate) {
+    return (
+      <>
```

Comment:
I guess you want me to use `?` operator to do conditional rendering? I just find if's more readable and since there are a couple (isPrivate, isAuth, isFullscreen) conditions I thought maybe ifs would be better? We could join those conditions together. But it's up to you ;)